Powershell changes to support new environments like china,germany

epkalyanr opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
<!--
Using the latest Job SDK(2.0.4) to support new environments like china,germany. Passing the storage suffix to the SDK inorder to support these new environments.
-->

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

msftclas commented:
@epkalyanr,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

azuresdkci commented:
Can one of the admins verify this patch?

markcowl commented on src/ResourceManager/HDInsight/Commands.HDInsight/JobCommands/GetAzureHDInsightJobOutputCommand.cs:
```diff
@@ -133,25 +134,38 @@ private static string Convert(Stream stream)
 
         internal IStorageAccess GetDefaultStorageAccess(string resourceGroupName, string clusterName)
         {
+            var StorageAccountSuffix = "";
```
Note that DefaultContext.Environment contains the appropriate storage suffix for the current azure environment (Azure, MoonCake, Germany, Government, STack), why are you not usign this value?

epkalyanr replied:
Parsing the storage account is a standard way to obtain the suffix. This way we are not dependent on the test execution context and the same code can be ported elsewhere. Also, when the storageAccountSuffix is null or empty the SDK picks the default azure environment.

markcowl replied:
the storageaccountsuffix is set for all of the built-in environments always.  It has nothing to do with the test context.  Using this value allows being compatible with existing offline scenarios for storage (i.e. when the user does not have access to storage management for the storage acocunt, but does have a key).

shefaliv replied:
@epkalyanr, please make the suggested change. This information is provided to us as part of the environment. We don't have this context in the SDK, so we don't have any other option. Since we do have the context in PS, we should use that instead!

markcowl commented on src/ResourceManager/HDInsight/Commands.HDInsight/JobCommands/GetAzureHDInsightJobOutputCommand.cs:
```diff
@@ -133,25 +134,38 @@ private static string Convert(Stream stream)
                     DefaultContainer = wasbAccount.StorageContainerName;
                     DefaultStorageAccountName = wasbAccount.StorageAccountName;
                     DefaultStorageAccountKey = wasbAccount.StorageAccountKey;
+
```
Test coverage needed.

epkalyanr replied:
This is an existing cmdlet and I am not modfying the parameters/arguments to it. I think the existing tests should suffice.

markcowl replied:
You are changing its execution.  Is the added code covered in existing tests?

shefaliv replied:
+1. Can you please add a test for this?

cormacpayne commented:
@epkalyanr there has been no activity for over a week; we will close this PR at the end of the week if the comments above are not addressed

epkalyanr commented:
I will look into the comments this week

azuresdkci commented:
Can one of the admins verify this patch?

epkalyanr commented:
Closing this pull request. Raised a new pull request addressing the comments made in this PR. Please find a link to the new PR below.
https://github.com/Azure/azure-powershell/pull/4495
